src/core/obj.js

```diff
@@ -32,578 +32,600 @@ function fetchDestination(dest) {
   return isDict(dest) ? dest.get('D') : dest;
 }
 
-var Catalog = (function CatalogClosure() {
-  function Catalog(pdfManager, xref) {
+class Catalog {
+  constructor(pdfManager, xref) {
     this.pdfManager = pdfManager;
     this.xref = xref;
+
     this.catDict = xref.getCatalogObj();
     if (!isDict(this.catDict)) {
-      throw new FormatError('catalog object is not a dictionary');
+      throw new FormatError('Catalog object is not a dictionary.');
     }
 
     this.fontCache = new RefSetCache();
     this.builtInCMapCache = new Map();
     this.pageKidsCountCache = new RefSetCache();
   }
 
-  Catalog.prototype = {
-    get metadata() {
-      var streamRef = this.catDict.getRaw('Metadata');
-      if (!isRef(streamRef)) {
-        return shadow(this, 'metadata', null);
-      }
+  get metadata() {
+    const streamRef = this.catDict.getRaw('Metadata');
+    if (!isRef(streamRef)) {
+      return shadow(this, 'metadata', null);
+    }
 
-      var encryptMetadata = (!this.xref.encrypt ? false :
+    const encryptMetadata = (!this.xref.encrypt ? false :
                              this.xref.encrypt.encryptMetadata);
-
-      var stream = this.xref.fetch(streamRef, !encryptMetadata);
-      var metadata;
-      if (stream && isDict(stream.dict)) {
-        var type = stream.dict.get('Type');
-        var subtype = stream.dict.get('Subtype');
-
-        if (isName(type, 'Metadata') && isName(subtype, 'XML')) {
-          // XXX: This should examine the charset the XML document defines,
-          // however since there are currently no real means to decode
-          // arbitrary charsets, let's just hope that the author of the PDF
-          // was reasonable enough to stick with the XML default charset,
-          // which is UTF-8.
-          try {
-            metadata = stringToUTF8String(bytesToString(stream.getBytes()));
-          } catch (e) {
-            if (e instanceof MissingDataException) {
-              throw e;
-            }
-            info('Skipping invalid metadata.');
+    const stream = this.xref.fetch(streamRef, !encryptMetadata);
```

Comment:
You're obviously just touching old code here, but this looks a bit convoluted. How about, in a separate commit, updating this to be a bit easier to read?

```javascript
const suppressEncryption = !(this.xref.encrypt &&
                             this.xref.encrypt.encryptMetadata);
const stream = this.xref.fetch(streamRef, suppressEncryption);
```